### PR TITLE
completed CRUD API

### PR DESCRIPTION
**Pull request by jiayitang0106**

### CODE REVIEW

File: client/components/App.jsx [hunk 1]
```diff
@@ -221,6 +221,7 @@ class App extends Component {
     const urlString = window.location.href;
     const url = new URL(urlString);
     const restaurantID = url.searchParams.get('restaurantID');
+    console.log("restaurant ID" + restaurantID);
```
Comment (dclay1983) [inline client/components/App.jsx]: Always remove or comment debugging logs when done with them.

File: server/db/index.js [hunk 2]
```diff
@@ -25,6 +25,29 @@ const get = (id) => {
...
+};
+
+const updateData = (id, data) => {
+    return new Promise((resolve, reject) => {
+      Restaurants.update({id}, data)
+        .exec((err, data) => {
+          if (err) {
+            reject(err);
+          }
+          resolve(data);
```
Comment (dclay1983) [inline server/db/index.js:39]: You assigned native promises to `mongoose.Promise`, therefore, you do not need to build a promise. Instead `return Restaurants.update({id}, data).then(data => doSomething).catch(err => doSomething)`

File: server/index.js [hunk 3]
```diff
@@ -24,6 +24,31 @@ app.get('/api/images/:restaurantId', (req, res) => {
...
+
+app.put('/api/images/:restaurantId', (req, res) => {
+  const data = req.body;
+  const id = req.params.restaurantId;
+  console.log("=============");
+  console.log(data);
+  console.log(id);
+  db.updateData(id, data)
+    .then(res.status(201).end())
+    .catch(error => console.log('Error occurs in Put'))
```
Comment (dclay1983) [inline server/index.js]: If an error occurs, the client still needs a res sent. Sometimes the error occurs just due to a timeout and cannot be detected during development. Your client will break if you do not return a response in a set time.

File: server/index.js [hunk 4]
```diff
@@ -24,6 +24,31 @@ app.get('/api/images/:restaurantId', (req, res) => {
...
+    .then(res.status(201).end())
+    .catch(err => console.log('Error occurs in Post'))
+});
+
+app.put('/api/images/:restaurantId', (req, res) => {
+  const data = req.body;
+  const id = req.params.restaurantId;
+  console.log("=============");
+  console.log(data);
+  console.log(id);
```
Comment (dclay1983) [inline server/index.js]: Logging database changges is good practice, but should be kept to 1 line. Something like: `Database: ${id} updated at ${new Date()}`

**Review by dclay1983 — CHANGES_REQUESTED** [review]
bundle.js should be included in your .gitignore. This should not be committed.

File: client/components/App.jsx (hunk 1 above)
Comment (jiayitang0106) [reply client/components/App.jsx]: Done, thanks.

File: server/index.js (hunk 3 above)
Comment (jiayitang0106) [reply server/index.js]: You are right. It makes sense. Just modified it. Thanks!

**Pull request closed without merging**